tests/pact/fqns.repl

```diff
@@ -77,3 +77,74 @@
 
 (expect "selects correct test" (modB.get-test) "hello")
 (commit-tx)
+
+;;
+;; Module redeploy name resolution
+;;
+
+(begin-tx)
+(namespace 'free)
+(module test-mod-redeploy-ref g
+  (defcap g () true)
+
+  (defcap test ()
+    (enforce false "boom"))
+
+  (defun f ()
+    (with-capability (test)
+      1))
+  )
+(expect-failure "Execution fails due to the falsy defcap" (f))
```

Comment:
I think `"Execution expected to fail due to assertion failure within defcap invoke - invoking with an unqualified reference."` is a better string.